src/search.js

```diff
@@ -0,0 +1,104 @@
+var _ = require('underscore');
+var config = require('./config');
+var bsonUrlEncoding = require('./utils/bsonUrlEncoding');
+
+/**
+ * Performs a search query on a Mongo collection and pages the results. This is different from
+ * find() in that the results are ordered by their relevancy, and as such, it does not take
+ * a paginatedField parameter. Note that this is less performant than find() because it must
+ * perform the full search on each call to this function. Also note that results might change
+ *
+ * @param {MongoCollection} collection A collection object returned from the MongoDB library's
+ *    `db.collection(<collectionName>)` method. This MUST have a Mongo $text index on it.
+ *    See https://docs.mongodb.com/manual/core/index-text/.
+ * @param {String} searchString String to search on.
+ * @param {Object} params
+ *    -query {Object} The find query.
+ *    -limit {Number} The page size. Must be between 1 and `config.MAX_LIMIT`.
+ *    -fields {Object} Fields to query in the Mongo object format, e.g. {title :1}.
+ *      The default is to query ONLY _id (note this is a difference from `find()`).
+ *    -next {String} The value to start querying the page. Defaults to start at the beginning of
+ *      the results.
```

Comment:
Nah, added that to 'future' since it's a bit tricky to implement with the aggregation.